### PR TITLE
[script.xbmc.unpausejumpback@matrix] 3.0.0

### PR DESCRIPTION
**Pull request by bossanova808**
### Add-on details:

- **General**
  - Add-on name: Unpause Jumpback
  - Add-on ID: script.xbmc.unpausejumpback
  - Version number: 3.0.0
  - Kodi/repository version: matrix

- **Code location**
  - URL: https://github.com/bossanova808/script.xbmc.unpausejumpback/
  

        This addon will jump back a number of seconds whenever a video is un-paused - to make sure you don't miss anything.
        You can set the amount of seconds to jump back, and the minimum duration of the pause before the addon will trigger a jump back.
        It also allows to jump back or forward after rewind or fast-forward of a specified amount of seconds to compensate for over-shooting.
        (Originally created by Memphiz, up-keep taken over by bossanova808 in 2020 for Kodi Matrix and beyond).
    

### Description of changes:

v3.0.0
        [new] Update for Kodi Matrix / Python 3, make unpause on resume vs. pause a choice, bugfixes and tidy up.
    

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

**Comment by bossanova808** [comment]
Ok this one is a bit different @enen92 

I have tried to contact the original author (Memphiz) here - https://github.com/Memphiz/script.xbmc.unpausejumpback/issues/13 - for the last couple of weeks.   There hasn't been a response.  

As explained there, I am happy to take it over and it seems _fairly_ clear (no commits, no forum responses in some time now) - that Memphiz has lost interest in this one.  But I believe he is team Kodi - perhaps you can get in touch with him?  Definitely not trying to tread on any toes or whatever, but I have formed the idea he is not interested in this anyway, and will (hopefully!) appreciate someone taking it over.

Because my city has just gone back on lockdown :( I had some time now (which I might not later) - so thought I'd get this PR going now, & here it is.

Thanks as ever for your attention/review.

**Comment by Memphiz** [comment]
@bossanova808 i have no idea in which ways you tried to contact me. I am pretty reachable via mail and GitHub IMO. Beside that I am fine and thankful that you want to take those over. I use this addon in my Setup and would not like to miss it tbh.

File: script.xbmc.unpausejumpback/addon.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.xbmc.unpausejumpback" name="Unpause Jumpback" version="3.0.0" provider-name="bossanova808,Memphiz,Lucleonhart,schumi2004,dkadioglu">
+  <requires>
+    <import addon="xbmc.python" version="3.0.0"/>
+  </requires>
+  <extension point="xbmc.service" library="default.py"></extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en_GB">Kodi automatic jump back on un-pausing of videos</summary>
+    <summary lang="bg">Превърта назад след пауза</summary>
```
Comment (enen92) [inline script.xbmc.unpausejumpback/addon.xml:9]: this should be bg_GB

File: script.xbmc.unpausejumpback/addon.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    <description lang="en_GB">
+        This addon will jump back a number of seconds whenever a video is un-paused - to make sure you don't miss anything.
+        You can set the amount of seconds to jump back, and the minimum duration of the pause before the addon will trigger a jump back.
+        It also allows to jump back or forward after rewind or fast-forward of a specified amount of seconds to compensate for over-shooting.
+        (Originally created by Memphiz, up-keep taken over by bossanova808 in 2020 for Kodi Matrix and beyond).
+    </description>
+    <description lang="bg">
```
Comment (enen92) [inline script.xbmc.unpausejumpback/addon.xml:16]: same

File: script.xbmc.unpausejumpback/resources/language/resource.language.en_gb/strings.po [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+msgid "Actually do jump back at resume (on), or pause (off)"
+msgstr ""
+
+msgctxt "#32042"
+msgid "(For low power systems, jump back on pause may behave more smoothly)"
+msgstr ""
+
+msgctxt "#32043"
+msgid "Also jump back on playback started?"
+msgstr ""
```
Comment (enen92) [inline script.xbmc.unpausejumpback/resources/language/resource.language.en_gb/strings.po:106]: missing new line at the end of the file

File: script.xbmc.unpausejumpback/resources/lib/common.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    log(message, exception_instance, level=xbmc.LOGNOTICE)
+
+
+def footprints(startup=True):
+    """
+    Log the startup of an addon, and key Kodi details that are helpful for debugging
+
+    :param startup: optional, default True.  If true, log the startup of an addon, otherwise log the exit.
+    """
+    if startup:
```
Comment (enen92) [inline script.xbmc.unpausejumpback/resources/lib/common.py:63]: all this belongs to debug logging not info

File: script.xbmc.unpausejumpback/resources/lib/common.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+def log_notice(message, exception_instance=None):
+    """
+    Log a message at the LOGNOTICE level, i.e. even if Kodi debugging is not turned on. Use sparingly.
+
+    :param message: required, the message to log
+    :param exception_instance: optional, an instance of some Exception
+    """
+
+    log(message, exception_instance, level=xbmc.LOGNOTICE)
```
Comment (enen92) [inline script.xbmc.unpausejumpback/resources/lib/common.py:54]: careful as lognotice was removed in matrix: https://codedocs.xyz/xbmc/xbmc/group__python__xbmc.html#ga3cd3eff2195727144ad6fbabd4744961

File: script.xbmc.unpausejumpback/resources/lib/unpause_jumpback.py [hunk 6]
```diff
@@ -0,0 +1,272 @@
+# -*- coding: utf-8 -*-
+
+import xbmc
+import time
+from resources.lib.common import *
```
Comment (enen92) [inline script.xbmc.unpausejumpback/resources/lib/unpause_jumpback.py:5]: you import all and then log

File: script.xbmc.unpausejumpback/resources/lib/unpause_jumpback.py [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+from resources.lib.common import log
+
+global g_jumpBackOnResume
```
Comment (enen92) [inline script.xbmc.unpausejumpback/resources/lib/unpause_jumpback.py:8]: all these globals are better as class members

**Comment by bossanova808** [comment]
@memphiz just via that issue on the repo I opened. Glad you are well!  And glad you don't mind me taking it over.

Thanks and look forward to the review @enen92

File: script.xbmc.unpausejumpback/addon.xml (hunk 1 above)
Comment (bossanova808) [reply script.xbmc.unpausejumpback/addon.xml:9]: Happy to change it of course, but just for my understanding... I always assumed that en_GB was for English, with a 'class' of British English essentially.  Is there an equivalent for Bulgarian then?

Comment (bossanova808) [reply script.xbmc.unpausejumpback/addon.xml:9]: Ah I think I see - maybe `bg_BG`?  not `bg_GB` ?

**Comment by bossanova808** [comment]
Ok @enen92 thanks again, those should all be done...

**Pull request merged**